modules/Battle.py

```diff
@@ -0,0 +1,834 @@
+            battler_indices = [int.from_bytes(ReadSymbol('gBattlerPartyIndexes', size=12)[2*i:2*i+2], 'little') for i in range(len(GetParty()))]
+            # this tells us how many pokemon are battling (2 for single battle, 4 for double) which allows us to get the pokemon info for the party members currently participating in the battle
+            num_battlers = ReadSymbol('gBattlersCount', size=1)[0]
+            # If we only have one party member, it's obviously the current battler so don't do the calcs for who's in battle
+            if len(GetParty()) == 1:
+                return GetParty()
+            current_battlers = [GetParty()[battler_indices[i]] for i in range(len(battler_indices)//num_battlers)]
+            return current_battlers
+
+
+def GetStrongestMove() -> int:
+    """
+    Function that determines the strongest move to use given the current battler and the current
+    """
+    current_battlers = GetCurrentBattler()
+    if len(current_battlers) > 1:
+        console.print("Double battle detected, feature not yet implemented.")
+        os._exit(2)
+    else:
+        current_battler = current_battlers[0]
+        move = FindEffectiveMove(current_battler, GetOpponent())
+        if move['power'] == 0:
+            console.print('Lead Pokémon has no effective moves to battle the foe!')
+            return -1
+
+        console.print('Best move against foe is {} (Effective power is {:.2f})'.format(
+            move['name'],
+            move['power']
+        ))
+        return move['index']
+
+
+def GetMonToSwitch(active_mon: int) -> int:
+    """
+    Figures out which pokemon should be switched out for the current active pokemon.
+
+    :param active_mon: the party index of the pokemon that is being replaced.
+    :return: the index of the pokemon to switch with the active pokemon
+    """
+    party = GetParty()
+    match config_battle['switch_strategy']:
+        case "first_available":
+            for i in range(len(party)):
+                if party[i] == active_mon:
+                    continue
+                # check to see that the party member has enough HP to be subbed out
+                elif party[i]['stats']['hp'] / party[i]['stats']['maxHP'] > .2:
+                    console.print('Pokémon {} has more than 20% hp!'.format(party[i]['name']))
+                    for move in party[i]['moves']:
+                        if (
+                                move['power'] > 0 and
+                                move['remaining_pp'] > 0 and
+                                move['name'] not in config_battle['banned_moves']
+                                and move['kind'] in ['Physical', 'Special']):
+                            console.print('Pokémon {} has usable moves!'.format(party[i]['name']))
+                            return i
+            console.print("Error finding new battler- party seems to be out of PP/HP. Stopping...")
```

Comment:
Can we please make it so that instead of completely stopping the bot, just set `config_battle['battle']` to `False` and continue shiny hunting?